src/reactionsystem_conversions.jl

```diff
@@ -89,41 +111,52 @@ function assemble_oderhs(rs, ispcs; combinatoric_ratelaws = true, remove_conserv
     rhsvec
 end
 
-function assemble_drift(rs, ispcs; combinatoric_ratelaws = true, as_odes = true,
-        include_zero_odes = true, remove_conserved = false)
-    rhsvec = assemble_oderhs(rs, ispcs; combinatoric_ratelaws, remove_conserved)
+# For a `ReactionSystem`, creates a vector with all the differential equations corresponding to
+# the ODEs generated by the reaction rate equation. The `as_odes` argument determine whether the LHS should
+# be differential (when generating e.g. a `ODESystem`) or `0` (when generating e.g. a NonlinearSystem).
+function assemble_drift(rs, ind_sps; combinatoric_ratelaws = true, as_odes = true,
+                        include_zero_odes = true, remove_conserved = false)
+    rhsvec = assemble_oderhs(rs, ind_sps; combinatoric_ratelaws, remove_conserved)
     if as_odes
         D = Differential(get_iv(rs))
         eqs = [Equation(D(x), rhs)
-               for (x, rhs) in zip(ispcs, rhsvec) if (include_zero_odes || (!_iszero(rhs)))]
+               for (x, rhs) in zip(ind_sps, rhsvec) if (include_zero_odes || (!_iszero(rhs)))]
```

Comment:
Thought `ind_sps` was when I went through, but feel free to tell me to revert if you do not agree.